src/lib/common/ostream.hpp

```diff
@@ -135,50 +140,51 @@ namespace details {
  */
 namespace std {
     //! @brief Printing arrays.
-    template <typename T, size_t n>
-    std::ostream& operator<<(std::ostream& o, const std::array<T, n>& v) {
+    template <typename O, typename T, size_t n>
```

Comment:
The definition of every `operator<<` should be guarded by a `if_ostream<0>` check, otherwise they will collide with serialising (are all tests really passing? I would be very suprised, and if so we probably need to add some tests).
You should be able to do it adding a further template argument `typename = if_ostream<O>` at the end of the list.